src/manifold/src/face_op.cpp

```diff
@@ -35,10 +40,130 @@ void Manifold::Impl::Face2Tri(const VecDH<int>& faceEdge,
   VecDH<glm::vec3> triNormal;
   VecDH<TriRef>& triRef = meshRelation_.triRef;
   triRef.resize(0);
-  triVerts.reserve(halfedge_.size());
-  triNormal.reserve(halfedge_.size());
-  triRef.reserve(halfedge_.size());
+#if MANIFOLD_PAR == 'T' && __has_include(<tbb/tbb.h>)
+  tbb::task_group group;
+  // map from face to triangle
+  tbb::concurrent_unordered_map<int, std::vector<glm::ivec3>> results;
+  VecDH<int> triCount(faceEdge.size());
+  triCount.back() = 0;
+  // precompute number of triangles per face, and launch async tasks to
+  // triangulate complex faces
+  for_each_host(
+      autoPolicy(faceEdge.size()), countAt(0), countAt(faceEdge.size() - 1),
+      [&](int face) {
+        triCount[face] = faceEdge[face + 1] - faceEdge[face] - 2;
+        ASSERT(triCount[face] >= 1, topologyErr,
+               "face has less than three edges.");
+        if (triCount[face] > 2)
+          group.run([&, face] {
+            const glm::vec3 normal = faceNormal_[face];
+            const glm::mat3x2 projection = GetAxisAlignedProjection(normal);
+            const PolygonsIdx polys = Face2Polygons(face, projection, faceEdge);
+            std::vector<glm::ivec3> newTris = TriangulateIdx(polys, precision_);
```

Comment:
Is this now calling `Triangulate` on even 3 and 4 sided polygons? We were separating those out before to reduce overhead since the vast majority of the polygons are those and don't require the full triangulator.

Reply:
No, we are skipping those with less than 5 sides.